File: frontend/src/app/features/pedidos/domain/interfaces/repositories/ICategoriaRepository.ts
import { Categoria } from '../../entities/categoria';

/**
 * Contrato del repositorio de categorías de productos.
 * @remarks Implementado por {@link CategoriaRepository}.
 */
export interface ICategoriaRepository {
  /** Devuelve todas las categorías. */
  getAll(): Promise<Categoria[]>;
  /**
   * Obtiene una categoría por su identificador.
   * @param id Identificador de la categoría.
   * @returns Entidad {@link Categoria} o `null` si no existe.
   */
  getById(id: number): Promise<Categoria | null>;
  /** Devuelve únicamente las categorías activas. */
  getActivos(): Promise<Categoria[]>;
}

File: backend/ERP.Backend/Domain/Interfaces/IRepository/ICategoriaRepository.cs
using Domain.Entities;

namespace Domain.Interfaces
{
    /// <summary>
    /// Interfaz del repositorio de categorías de producto.
    /// Define las operaciones de lectura sobre <see cref="CategoriaProducto"/>.
    /// </summary>
    public interface ICategoriaRepository
    {
        /// <summary>
        /// Obtiene todas las categorías de producto.
        /// </summary>
        /// <returns>Lista de <see cref="CategoriaProducto"/> con todos los registros.</returns>
        Task<List<CategoriaProducto>> GetAllAsync();

        /// <summary>
        /// Obtiene una categoría por su identificador único.
        /// </summary>
        /// <param name="categoriaID">Identificador único de la categoría a buscar.</param>
        /// <returns>La <see cref="CategoriaProducto"/> encontrada, o <c>null</c> si no existe.</returns>
        Task<CategoriaProducto?> GetByIdAsync(int categoriaID);
    }
}